lib/npm.js

```diff
@@ -0,0 +1,132 @@
+import _ from 'lodash';
+import path from 'path';
+import { exec } from 'teen_process';
+import { mkdirp, util } from 'appium-support';
+
+const INSTALL_LOCKFILE = '.appium.install.lock';
+const LINK_LOCKFILE = '.appium.link.lock';
+
+export default class NPM {
+
+  constructor (appiumHome) {
+    this.appiumHome = appiumHome;
+  }
+
+  async exec (cmd, args, opts, execOpts = {}) {
+    let { cwd, json, lockFile } = opts;
+    if (!cwd) {
+      cwd = this.appiumHome;
+    }
+    // ensure the directory we want to install inside of exists
+    await mkdirp(cwd);
+    execOpts.cwd = cwd;
+
+    args.unshift(cmd);
+    if (json) {
+      args.push('-json');
+    }
+    let runner = async () => await exec('npm', args, execOpts);
+    if (lockFile) {
+      const acquireLock = util.getLockFileGuard(path.resolve(cwd, lockFile));
+      const _runner = runner;
```

Comment:
why this is needed?

Reply:
because if we don't alias this, then we end up with a recursive function which is not what we want.